src/modules/decentralized-web-node/services/handler.service.ts

```diff
@@ -0,0 +1,143 @@
+import { Injectable } from '@nestjs/common';
+import { plainToClass } from 'class-transformer';
+import { validate } from 'class-validator';
+import { CID } from 'multiformats/cid';
+import * as json from 'multiformats/codecs/json';
+import { sha256 } from 'multiformats/hashes/sha2';
+import { Logger } from '../../logger/logger.service';
+import { CollectionsQueryDescriptorDto } from '../dtos/collections-query-descriptor.dto';
+import { CollectionsWriteDescriptorDto } from '../dtos/collections-write-descriptor.dto';
+import { MessageDto } from '../dtos/message.dto';
+import { ResponseReplyDto } from '../dtos/response-reply.dto';
+import { CollectionsWriteMessage } from '../entities/collections-write-message.entity';
+import { MessageService } from './message.service';
+
+@Injectable()
+export class CollectionHandlerService {
+  constructor(
+    private readonly logger: Logger,
+    private readonly messageService: MessageService
```

Comment:
Maybe it's just me, but I find the naming/relationship between this class and `MessageService` to be pretty confusing. I guess we should start by changing the name of this class to align with the filename.

In your view, why are the `HandlerService` and `MessageService` separate? It is nice to have smaller classes but the distinction between them isn't really clear to me

```suggestion
export class HandlerService {
  constructor(
    private readonly logger: Logger,
    private readonly messageService: MessageService
```

Reply:
So, `MessageService` has a method that allow you to modify and work with `CollectionsWriteMessage` entities. It's more like repository I'd say. On the other hand, we have a service called `HandlerService` that is responsible just for receiving query/write messages and manage them.

Reply:
Ok, thanks for the explanation. We can leave as is, I think it's better to have them separate than together 👍

Reply:
It's way readable and we can only share a `HandlerService` between modules, so no direct operation on entities, only via query/write message.